pkg/splunk/enterprise/util.go

```diff
@@ -913,16 +913,26 @@ func changePhaseInfo(desiredReplicas int32, appSrc string, appSrcDeployStatus ma
 				continue
 			}
 
+			// CSPL-1570: Set the AuxPhaseInfo to Install complete for all the existing pods.
+			// This is done so that we don't try to reinstall the app again on the existing pod.
+			if len(appDeployInfoList[idx].AuxPhaseInfo) == 0 &&
```

Comment:
To make the scale-up completely decoupled from the scheduler, Please remove this check as well. Then, at line 922, we can reset the slice, OR, update the existing slice. 

Also, looking at the overall flow, one another case is(rare case, scale-up, when in the middle of installation),  if the Main phase info is not Install complete:
1. Do not alter the existing auxPhaseInfo
2. Still set the PodCopy phase for new pod entries.


I feel, with these changes, it will be independent, and self-sufficient with respect to the scale-up scenario.  With that, the scheduler will get the calls in 3 different flows, and all of them are handled in the same way, and no special handle for any of these cases. i.e (1. First-time entry, once after learning new apps, 2. Re-entry across reconciles, 3. scale-up case.)

Reply:
Done